BrightID-rn/src/store/index.js

```diff
@@ -1,10 +1,50 @@
 //store/index.js
 import thunkMiddleware from "redux-thunk";
-import { createStore, applyMiddleware } from "redux";
+import {createStore, applyMiddleware} from "redux";
 import reducer from "../reducer";
+import {setPPKeys} from "../actions/index";
+import nacl from "tweetnacl"
+import axios from "axios"
 
 const store = createStore(reducer, applyMiddleware(thunkMiddleware));
 
+export const setupPPKeys = ppKeys => async dispatch => {
+
+    try {
+
+        if (ppKeys.hasOwnProperty("key1") && ppKeys.hasOwnProperty("key2")) {
+            dispatch(setPPKeys(ppKeys));
+        }
+    } catch (err) {
+        console.error(err);
+    }
+};
+
+export const generatePPKeys = () => async dispatch => {
+
+    try {
+
+        key1 = nacl.box.keyPair();
+        key2 = nacl.box.keyPair();
+
+        axios.post("http://rest.learncode.academy/api/brightid/keys",
+            {"key1": key1.publicKey, "key2": key2.publicKey},
+            {'Content-Type': 'application/json'})
+            .then((response) => {
+                if (response === "ok") {
+                    dispatch(setPPKeys({"key1": key1, "key2": key2}));
```

Comment:
At this point we should also store the keys in AsyncStorage. in an encrypted format.

Reply:
yes i agree we should store the keys in an encrypted format, everything we place into async storage is a string so it shouldn't be that difficult to implement local encyption

Reply:
My feeling about encrypting the private/public keys is that we should do it, but not for the MVP/beta.  Especially when people know what BrightID is (and might try to steal someone's private key) or we want keys to be portable (for transferring your ID to another device for example), this will become important.  This will probably be one the first features after the beta release.  I created [an issue](https://github.com/BrightID/BrightID/issues/73) for it under ["Beyond MVP"](https://github.com/BrightID/BrightID/projects/4).